parcels_benchmarks/benchmark_setup.py

```diff
@@ -0,0 +1,264 @@
+    dataset_folder : Path
+        Path to the folder containing the downloaded dataset files.
+    """
+    # Dev note: `dataset` is assumed to be a folder name with netcdf files
+    if dataset not in DATA_FILES:
+        raise ValueError(
+            f"Dataset {dataset!r} not found. Available datasets are: " + ", ".join(DATA_FILES.keys())
+        )
+    odie = _get_pooch(data_home=data_home)
+    print(f"dataset: {dataset}")
+    print(f"Fetching: {DATA_FILES[dataset]}")
+    listing = odie.fetch(DATA_FILES[dataset],processor=pooch.Unzip())
+
+    # as pooch currently returns a file listing while we want a dir,
+    # we find the common parent dir to all files
+    common_parent_dir = min([Path(f) for f in listing], key=lambda f: len(f.parents)).parent
+
+    return common_parent_dir
+
+def download_datasets(data_home=None):
+    """Download all datasets listed in the config file to the specified location.
+
+    Parameters
+    ----------
+    data_home : pathlike, optional
+        The directory in which to cache data. If not specified, the value
+        of the ``PARCELS_EXAMPLE_DATA`` environment variable, if any, is used.
+        Otherwise the default location is assigned by :func:`get_data_home`.
+
+    Returns
+    -------
+    dataset_folders : dict
+        Mapping of dataset names to paths to the folder containing the downloaded dataset files.
+    """
+    dataset_folders = {}
+    for dataset in DATA_FILES:
+        folder = download_example_dataset(dataset, data_home=data_home)
+        dataset_folders[dataset] = folder
+    return dataset_folders
+
+
+def retrieve_data_dir(
+    url: str = None,
+    known_hash: str = None,
+):
+    # let pooch retrieve and get a listing of all included files
+    listing = pooch.retrieve(
+        url,
+        processor=pooch.Unzip(),
+        known_hash=known_hash,
+    )
+
+    # as pooch currently returns a file listing while we want a dir,
+    # we find the common parent dir to all files
+    common_parent_dir = min([Path(f) for f in listing], key=lambda f: len(f.parents)).parent
+
+    return str(common_parent_dir)
+
+
+def list_datasets():
+    """Print available dataset names and their files."""
+    for name, files in DATA_FILES.items():
+        print(f"{name}:")
+        for f in files:
+            print(f"  - {f}")
+
+
+def dataset_cache_root(data_home=None) -> Path:
+    """Return the root cache directory used by pooch."""
+    odie = _get_pooch(data_home=data_home)
+    return Path(odie.path)
+
+
+def dataset_paths(dataset: str, data_home=None) -> list[Path]:
+    """Return full paths to all files for a given dataset (if downloaded)."""
+    if dataset not in DATA_FILES:
+        raise ValueError(
+            f"Unknown dataset {dataset!r}. "
+            f"Available datasets: {', '.join(DATA_FILES.keys())}"
+        )
+    root = dataset_cache_root(data_home=data_home)
+    return [root / fname for fname in DATA_FILES[dataset]]
+
+
+def cli() -> argparse.ArgumentParser:
```

Comment:
I'm having a hard time understanding the CLI.

1. The different arguments and flags are unclear to me.

2. If I use it with the `--data-home ./mydatahome/ "download"`. How is it ensured that the benchmark run then actually looks for the data in this location?